contrib/src/json.rs

```diff
@@ -88,18 +88,23 @@ impl<T> Json<T> {
 const LIMIT: u64 = 1 << 20;
 
 impl<T: DeserializeOwned> FromData for Json<T> {
-    type Error = SerdeError;
+    type Error = io::Error;
 
-    fn from_data(request: &Request, data: Data) -> data::Outcome<Self, SerdeError> {
+    fn from_data(request: &Request, data: Data) -> data::Outcome<Self, io::Error> {
         if !request.content_type().map_or(false, |ct| ct.is_json()) {
             error_!("Content-Type is not JSON.");
             return Outcome::Forward(data);
         }
 
         let size_limit = request.limits().get("json").unwrap_or(LIMIT);
-        serde_json::from_reader(data.open().take(size_limit))
+        let mut buf = String::new();
```

Comment:
Could you use `with_capacity` here with `size_limit` to optimize this a little more?

Reply:
We don't want to over-allocate memory. Perhaps we could choose a safe lower-bound: something like 1k or 4k.